scripts/fetchODCDataFilesUtils.js

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2020 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in compliance with this licence.
+You may obtain a copy of the BSD License at
+https://raw.githubusercontent.com/inclusive-design/data-update-github/main/LICENSE
+*/
+
+// Utility functions used by fetchODCDataFiles.js
+
+"use strict";
+
+const fs = require("fs");
+const axios = require("axios");
+const JSDOM = require("jsdom").JSDOM;
+const rimraf = require("rimraf");
+const git = require("simple-git")();
+
+module.exports = {
+	/**
+	 * Scrape the download link and date of last update from the ODS repository page
+	 * @param {String} dataSourceURL The URL to the webpage where the information of the data file is published
+	 * @return {Object} An object keyed by "downloadUrl" (the url to download the new data file) and
+	 * "date" (the last updated date of the data file in a format of YYYY-MM-DD). These values are `undefined` when
+	 * the csv link is not found or any error happens.
+	 */
+	getDataSource: async (dataSourceURL) => {
+		try {
+			let res = await axios.get(dataSourceURL);
+			let dom = new JSDOM(res.data);
+
+			// find the CSV download link
+			let downloadLink, publishedDate;
+			let as = dom.window.document.querySelectorAll("a.dataset-download-link");
+			for (let a of as) {
+				const link = a.getAttribute("href");
+				if (link.slice(-4) === ".csv") {
+					downloadLink = link;
+					var dateElmContent = a.closest(".resource-item").querySelector(".description.details").innerHTML;
+					const matchStartStr = "Last Updated: ";
+					const matchEndStr = " |";
+					publishedDate = dateElmContent.substring(dateElmContent.lastIndexOf(matchStartStr) + matchStartStr.length, dateElmContent.lastIndexOf(matchEndStr));
+					publishedDate = module.exports.formatDate(publishedDate);
+					break;
```

Comment:
Rather than bailing out, I suggest this should be an as.filter() returning all matching links, and then a warning issued if we end up finding more than one - or is the idea that we expect to use only the first of any matching set?

Reply:
I will improve the code to report an error when more than one csv download links are found or no csv link is found.